src/modules/collections/controllers/collections.controller.ts

```diff
@@ -70,13 +74,38 @@ export class CollectionsController {
 		@Query() queryDto: CollectionObjectsQueryDto,
 		@SessionUser() user: SessionUserEntity
 	): Promise<IPagination<IeObject>> {
-		const collectionObjects = await this.collectionsService.findObjectsByCollectionId(
+		const folderObjects = await this.collectionsService.findObjectsByCollectionId(
 			collectionId,
 			user.getId(),
 			queryDto,
 			referer
 		);
-		return collectionObjects;
+		const maintainerIds = folderObjects.items.map((item) => item.maintainerId);
+		const visits = await this.visitsService.findAll(
+			{ size: 1000, status: VisitStatus.APPROVED, timeframe: VisitTimeframe.ACTIVE },
+			{
+				userProfileId: user.getId(),
+				visitorSpaceStatus: VisitorSpaceStatus.Active, // a visitor should only see visits for active spaces
+			}
+		);
+		const showThumbnailByMaintainerId = Object.fromEntries(
+			maintainerIds.map((maintainerId) => {
+				const hasAccessToSpace = !!visits.items.find(
+					(item) => item.spaceMaintainerId.toLowerCase() === maintainerId.toLowerCase()
+				);
+				return [maintainerId, hasAccessToSpace];
+			})
+		);
+
+		// Redact folder objects based on, if the user has access to that space at the moment
+		folderObjects.items.forEach((folderObject) => {
+			if (showThumbnailByMaintainerId[folderObject.maintainerId]) {
+				return;
+			}
+			// Redact object
+			delete folderObject.thumbnailUrl;
+		});
```

Comment:
This has a lot of overlap with `src/modules/media/controllers/media.controller.ts:298`'s `applyLicensesToObject`. Can we centralise this everywhere we return an item?